docs/pubsub-cloudevents.md

```diff
@@ -0,0 +1,374 @@
+compliant media-type expression.
+
+### 2.2. data Attribute
+
+The `data` attribute is assumed to contain opaque application data that is
+encoded as declared by the `datacontenttype` attribute.
+
+An application is free to hold the information in any in-memory representation
+of its choosing, but as the value is transposed into a Pub/Sub message as defined in this
+specification, the assumption is that the `data` attribute value is made
+available as a sequence of bytes.
+
+For instance, if the declared `datacontenttype` is
+`application/json;charset=utf-8`, the expectation is that the `data` attribute
+value is made available as [UTF-8][rfc3629] encoded JSON text Pub/Sub message data.
+
+## 3. Pub/Sub Message Mapping
+
+The content mode is chosen by the sender of the event.
+
+The receiver of the event can distinguish between the three modes by inspecting
+the `Content-Type` attribute value. If the value is prefixed with the CloudEvents
+media type `application/cloudevents`, indicating the use of a known
+[event format](#14-event-formats), the receiver uses _structured_ mode. If the
+value is prefixed with `application/cloudevents-batch`, the receiver uses the
+_batched_ mode. Otherwise it defaults to _binary_ mode.
+
+If a receiver detects the CloudEvents media type, but with an event format that
+it cannot handle, for instance `application/cloudevents+avro`, it MAY still
+treat the event as binary and forward it to another party as-is.
+
+### 3.1. Binary Content Mode
+
+The _binary_ content mode accommodates any shape of event data, and allows for
+efficient transfer and without transcoding effort.
+
+#### 3.1.1. Content-Type
+
+For the _binary_ mode, the attribute `Content-Type` value is not used.
+
+#### 3.1.2. Event Data Encoding
+
+The [`data` attribute](#22-data-attribute) byte-sequence is used as the Pub/Sub
+message data payload.
+
+#### 3.1.3. Metadata Attributes
+
+All [CloudEvents][ce] attributes 
+MUST be individually mapped to and from distinct Pub/Sub message attributes, with
+exceptions noted below.
+
+CloudEvents extensions that define their own attributes MAY define a diverging
+mapping to Pub/Sub attributes for those attributes, especially if specific attributes
+need to align with Pub/Sub features or with other specifications that have explicit
+Pub/Sub attribute bindings.
+
+An extension specification that defines a diverging mapping rule for Pub/Sub, and
+any revision of such a specification, MUST also define explicit mapping rules
+for all other transport bindings that are part of the CloudEvents core at the
+time of the submission or revision.
+
+##### 3.1.3.1 Pub/Sub Attribute Names
+
+Except for attributes
+[explicitly handled in this specification](#2-use-of-cloudevents-attributes),
+the naming convention for the Pub/Sub attribute mapping of well-known CloudEvents
+attributes is that each attribute name MUST be prefixed with "ce-".
+
+Examples:
+
+    * `time` maps to `ce-time`
+    * `id` maps to `ce-id`
```

Comment:
Something to consider, how do the `time` and `id` attributes specified here relate to the `message_id` and `publish_time` fields on the Google [PubsubMessage](https://cloud.google.com/pubsub/docs/reference/rpc/google.pubsub.v1#pubsubmessage)? My thought is that they should be ignored, with the caveat that if something is receiving non cloud event messages through Google Pubsub, and they want to express these messages as cloud events, then those fields may be used to seed those values.

Reply:
Ah didn't see that. Thanks!